### PR TITLE
Changing The way Coreclr Test Builds

### PR DESCRIPTION
**Pull request by ramarag**
This PR removes all the individual project.json and replaces them with a single  tests\src\Common\test_dependencies\project.json which the tests can build against

@wtgodbe @dagood @rahku @jkotas PTAL

@dotnet/coreclr-contrib  FYI

### CODE REVIEW

File: tests/runtest.proj [hunk 1]
```diff
@@ -354,7 +352,7 @@ namespace $([System.String]::Copy($(Category)).Replace(".","_").Replace("\","").
   <!-- All the test projects need to add dependency to currently built runtime as they require that to run. 
        In addition the version number of built runtime can change so all project.json needs to be dynamically generated. 
        In order to avoid regenerating project.json for all the tests projects...I added a common project for all the tests.
-       This project is at tests\src\common\test_runtime\test_runtime.csproj. This project is adding common dependencies to
+       This project is at tests\src\common\test_dependencies\test_dependencies.csproj. This project is adding common dependencies to
        all the tests like dependency on Microsoft.netcore.runtime.coreclr ..etc. Project.json for this project is dynamically
```
Comment (rahku) [inline tests/runtest.proj]: This comment is now confusing. The target actually only refers to test_runtime directory but the comment refers to test_dependencies dir.

**Comment by rahku** [comment]
Can I still build an individual test without building whole test repo first?

**Comment by ramarag** [comment]
yes you can, i have not regressed any of the existing scenario (hopefully !!)

File: tests/runtest.proj (hunk 1 above)
Comment (ramarag) [inline tests/runtest.proj]: Yes you are right, the file referred here has not been used for the above purpose for a long time. will remove the file from the comment

File: tests/src/GC/Performance/Tests/project.json [hunk 2]
```diff
@@ -1,48 +0,0 @@
-{
-  "dependencies": {
```
Comment (ramarag) [inline tests/src/GC/Performance/Tests/project.json]: @Maoni0 @swgillespie  Do you guys really need a  seperate project.json ?

**Comment by ramarag** [comment]
@dotnet/jit-contrib i have removed all the project.json under tests\src\Jit\config and replaced them with the two project.json mentioned in this PR. IS there any use of keeping the seperate project.json for JIT ?

**Comment by ramarag** [comment]
@dotnet-bot test  Windows_NT x64 Debug Build and Test

**Comment by ramarag** [comment]
@dotnet-bot test Windows_NT x64 Debug Build and Test

**Comment by ramarag** [comment]
@dotnet-bot  test Windows_NT x86 legacy_backend Checked Build and Test

**Comment by wtgodbe** [comment]
@dotnet-bot test CentOS7.1 pri1 please

**Comment by wtgodbe** [comment]
@dotnet-bot test OpenSUSE13.2 pri1 please

**Comment by wtgodbe** [comment]
@dotnet-bot test Debian8.4 pri1 please

**Comment by wtgodbe** [comment]
@dotnet-bot test RHEL7.2 pri1 please

**Comment by gkhanna79** [comment]
CC @pgavlin @russellhadley @RussKeldorph for JIT changes

Overall, this is goodness. I am fine with this change so long as the PR is green and all P1 tests pass.

**Comment by pgavlin** [comment]
What does this buy us? In the past, I have found that there is quite a bit of value in having a more precise list of the dependencies for a given test. Does this improve test build times?

**Comment by CarolEidt** [comment]
I would prefer not to merge the jit project.json files. I am currently attempting to add support for reference overloads to the SIMD intrinsics, and it is already quite painful to test this. Eliminating any specialized project.json files will make it even more difficult.

**Comment by wtgodbe** [comment]
@dotnet-bot test CentOS7.1 pri1 please

**Comment by adityamandaleeka** [comment]
I haven't looked at all the files, but from what I've seen so far, your change leaves a ton of empty ItemGroup elements like this (which used to contain project.json and/or app.config):

```
   <ItemGroup>
    </ItemGroup>
```

**Comment by ramarag** [comment]
@pgavlin @CarolEidt I have removed the JIT side changes i made except for empty\project.json which i have moved up to be used by any project that wants to build against mscorlib facade

**Comment by ramarag** [comment]
@adityamandaleeka we have many such gunk left in the project files. The main motivation behind removing the two lines was to ensure no one authors a new project and expect to use them. The gunk that remain are harmless

**Comment by adityamandaleeka** [comment]
@ramarag I am aware that an empty ItemGroup doesn't have a functional impact, but since you're making a bulk change to over 14000 files, I figure you could modify whatever script you're using to remove the elements that are made redundant by your change.

In general, I don't think the fact that there may be some junk in the project files today is justification for adding more when it's easily avoidable.

**Comment by gkhanna79** [comment]
@ramarag It will be good to make things better with each change we make. In that light, is it possible to cleanup the empty ItemGroups @adityamandaleeka is talking about?

**Comment by ramarag** [comment]
removed the extra empty props from the project file

**Comment by ramarag** [comment]
@dotnet-bot test Debian8.4 pri1 please
@dotnet-bot test RHEL7.2 pri1 please

**Pull request merged**